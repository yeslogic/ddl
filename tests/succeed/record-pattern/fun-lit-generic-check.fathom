let fst = (fun (A: Type) (B: Type) ({x, y}: {x: A, y: B}) => x) : _;
let snd = (fun (A: Type) (B: Type) ({x, y}: {x: A, y: B}) => y) : _;
{}
