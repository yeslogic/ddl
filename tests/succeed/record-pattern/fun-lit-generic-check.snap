stdout = '''
let fst : fun (A : Type) (B : Type) -> { x : A, y : B } -> A =
fun A B _ => let x : A = _.x;
let y : B = _.y;
x;
let snd : fun (A : Type) (B : Type) -> { x : A, y : B } -> B =
fun A B _ => let x : A = _.x;
let y : B = _.y;
y;
() : ()
'''
stderr = ''
